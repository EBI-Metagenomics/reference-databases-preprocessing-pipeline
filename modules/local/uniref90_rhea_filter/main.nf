process UNIREF90_RHEA_FILTER {
    label 'process_single'
    container 'quay.io/biocontainers/pyfastx:2.2.0--py39h0699b22_0'

    input:
    path uniref90_fasta
    path uniprot_rhea_mapping

    output:
    path 'uniref90_with_rhea.fasta', emit: filtered_proteins
    path "versions.yml"            , emit: versions

    script:
    """
    uniref90_rhea_filter.py ${uniref90_fasta} ${uniprot_rhea_mapping} uniref90_with_rhea.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        biopython: \$(python -c "import importlib.metadata; print(importlib.metadata.version('biopython'))")
        uniref90_db: ${params.uniref90_version}
        uniprotKB_access_date: ${params.uniprotKB_access_date}
    END_VERSIONS
    """
}