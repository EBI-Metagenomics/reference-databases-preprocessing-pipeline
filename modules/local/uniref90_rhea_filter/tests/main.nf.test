nextflow_process {

    name "Test Process UNIREF90_RHEA_FILTER"
    script "../main.nf"
    process "UNIREF90_RHEA_FILTER"
    
    tag "uniref90"

    test("Run UNIREF90_RHEA_FILTER on uncompressed fasta") {

        when {
            params {
                uniref90_version = "2024_05"
                rheadb_version = "135"
                uniprotKB_access_date = "2024-07-31"
                outdir = "${outputDir}"
            }
            process {
                """
                input[0] = Channel.fromPath("${moduleDir}/tests/data/uniref90.test.fasta", checkIfExists: true)
                input[1] = Channel.fromPath("${moduleDir}/tests/data/uniprot_rhea_mapping.test.tsv", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Run UNIREF90_RHEA_FILTER on fasta.gz") {

        when {
            params {
                uniref90_version = "2024_05"
                rheadb_version = "135"
                uniprotKB_access_date = "2024-07-31"
                outdir = "${outputDir}"
            }
            process {
                """
                input[0] = Channel.fromPath("${moduleDir}/tests/data/uniref90.test.fasta.gz", checkIfExists: true)
                input[1] = Channel.fromPath("${moduleDir}/tests/data/uniprot_rhea_mapping.test.tsv", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}
