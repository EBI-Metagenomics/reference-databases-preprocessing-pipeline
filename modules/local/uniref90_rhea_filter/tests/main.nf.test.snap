{
    "Remove from protein fasta proteins without Rhea annotation": {
        "content": [
            {
                "0": [
                    "uniref90_with_rhea.fasta:md5,e0be31822a0d13f2a5dc1fb2afb59bd1"
                ],
                "1": [
                    "versions.yml:md5,7ccfd1ba7191faa2479162dd9296ce8c"
                ],
                "filtered_proteins": [
                    "uniref90_with_rhea.fasta:md5,e0be31822a0d13f2a5dc1fb2afb59bd1"
                ],
                "versions": [
                    "versions.yml:md5,7ccfd1ba7191faa2479162dd9296ce8c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-13T11:50:14.17324"
    }
}